src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipelineRequest.cs

```diff
@@ -29,6 +29,8 @@ public virtual void AddHeader(string name, string value)
 
         public abstract IEnumerable<HttpHeader> Headers { get; }
 
+        public abstract string RequestId { get; set; }
```

Comment:
Should it be typed as Guid? It would allow us to lazy allocate string

Reply:
We might end up allocating the same string multiple times when using it. RequestId is used multiple times in 99.9% of the pipelines so this string would always be allocated it's better to cache it.